testing/integration/package_version_test.go

```diff
@@ -130,13 +148,13 @@ func TestComponentBuildHashInDiagnostics(t *testing.T) {
 		allHealthy,
 		5*time.Minute, 10*time.Second,
 		"agent never became healthy. Last status: %v", &stateBuff)
-	t.Cleanup(func() {
+	defer func() {
```

Comment:
Why are we switching from `Cleanup` to `defer` here? As far as I'm aware these should be interchangeable if we aren't using subtests.